src/components/InterestForm/index.tsx

```diff
@@ -0,0 +1,83 @@
+'use client';
+
+import { useState } from 'react';
+import { UUID } from 'crypto';
+import { insertInterest } from '../../api/supabase/queries/interest';
+import { Interest, CaseListing } from '../../types/schema';
+import { Button } from '../Button';
+import {
+  FormContainer,
+  FormTextArea,
+  FormInput,
+  RadioGroup,
+  RadioLabel,
+  RadioInput,
+} from './styles';
+
+const radioOptions = ['Attorney', 'Interpreter', 'Both Attorney & Interpreter'];
+
+export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+  const [reason, setReason] = useState<string>('');
+  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [startDate, setStartDate] = useState<string>('');
+
+  const handleInsert = async () => {
+    if (reason !== '') {
+      const newInterest: Interest = {
+        // hardcoded values for now
+        id: crypto.randomUUID() as UUID,
+        listing_id: caseData.id,
+        listing_type: 0,
+        user_id: '36b8f84d-df4e-4d49-b662-bcde71a8764f',
+        form_response: {
+          interestType: 'IJP-6-test',
+          interestReason: reason,
+          start_date: startDate,
+        },
+      };
+      await insertInterest(newInterest);
```

Comment:
Not sure if this is just on my end, but for me this doesn't seem to be writing an interest to the table? All of this seems fine though, so I'm not sure why this might be 🤔 

But also since this sprint is entirely frontend-focused, we can look deeper into this in your next sprint!

Reply:
Yup, there's a 409 error rn... I'll look more into it during the next sprint to not delay merging the frontend changes in!